base/Base/Base_implementation.F90

```diff
@@ -10,11 +10,11 @@
   ! !USES:
   !
   use ESMF
-  use MAPL_ConstantsMod, only: MAPL_PI, MAPL_PI_R8,MAPL_DEGREES_TO_RADIANS
+  !use MAPL_Constants, only: MAPL_PI, MAPL_PI_R8,MAPL_DEGREES_TO_RADIANS
```

Comment:
I suppose this comment can be deleted?

Reply:
> > > Query: In file `MAPL_cfio/ESMF_CFIOUtilMod.F90` the following parameter can be found:
> > > ` PARAMETER (IGREG=15+31*(10+12*1582))` and
> > > `PARAMETER (IGREG=2299161)`
> > > These are the first day of the gregorian calendar and the julian day count of the initial gregorian calendar respectively. Thoughts on adding these to either InternalConstants.F90 or MathConstants.F90? Their usage is limited but I've seen some components using these.
> > 
> > 
> > Can you summarize where these are used (outside of MAPL_cfio)? The model itself should not be depending on such things I would hope.
> 
> Sure.
> I found usage of iGreg (defined every time) in:
> `Components/@GEOSgcm_GridComp/GEOSogcm_GridComp/GEOSocean_GridComp/GuestOcean_GridComp/MOM_GEOS5PlugMod/@mom/src/mom5/ocean_param/gotm-4.0/util/time.F90`
> 
> `Shared/MAPL/MAPL_cfio/ESMF_CFIOUtilMod.F90`
> 
> `Shared/MAPL/base/MAPL_NominalOrbitsMod.F90`
> 
> `Shared/GMAO_Shared/GMAO_etc/zeit_jday.f`
> 
> and
> 
> `Shared/GMAO_Shared/GMAO_gfio/timesubs.f`

The mom reference is odd, as mom does not use MAPL.   But the others are enough to mean that we can not get rid of them.  How does CFIO export both of these, as they have the same name?

Reply:
Deleted

Reply:
@tclune They are declared in different subroutines so they don't conflict with one another. What I was thinking was of adding two new variables to Constants `IGREG` or fully spelt out `INITIAL_GREGORIAN` and `JIGREG` or fully spelt out `JULIAN_INITIAL_GREGORIAN` if you think it is worth it so that other repos like GMAO_Shared could simply use our definitions from MAPL_Constant rather than them having to define it every single time. But if its not worth it as usage is not common, then this can be left as is.

Reply:
Ok - those IGREG are not used anywhere then - they are internal to the subroutines.   The other layers you pointed out must define their own values.

`MAPL_cfio` is  deprecated (mostly replaced by pfio, but not quite all the strings have been cut yet), so no point in fixing it.